docs/integrations/fetching-incidents.md

```diff
@@ -124,6 +124,69 @@ If you do not have any incidents to return then just return an empty list to ```
 demisto.incidents([])
 ```
 
+## Advenced
+
+### Generic fetch with look back
+Use the generic look back functions for fetching missing incidents that are indexed a while after they are created.
+
+#### Use case
+When a 3rd party product creates incidents, sometimes it might be indexed a while after it was created.
+For example, if incident "a" was created before incident "b", and just "b" was indexed, the fetch-incidents will fetch only "b" and will miss "a", also if "a" was indexed after the fetch were called, the next fetch will fetch only from the created time of "b".
+
+#### Solution
+Having the parameter look_back we can configure how nuch the fetch-incidents will look back in time (minutes) to get the incidents that were created a while ago but indexed a minute ago.
+
+The solution functions are in [CommonServerPython](https://xsoar.pan.dev/docs/reference/api/common-server-python) and you can find the exact documentation there, but here is a short description.
+
+* **get_fetch_run_time_range** - The function gets the last run object and some other arguments and calculate the time to fetch and returns the start and end time to fetch.
+If the parameter look_back is given, then the start time will always be >= than `now - look_back`.
+
+* **filter_incidents_by_duplicates_and_limit** - After getting the incidents from the 3rd party API call, we will want to filter the incidents from duplicates.
+In the example above, after incident "a" is indexed the fetch will bring incidents "a" and "b", then we want to return only "a" because "b" is already fetched.
+In addition, after filtering from duplicates, if we still have more incidents than the limit, the function will return only by the limit.
+
+* **get_latest_incident_created_time** - Given a list of incidents and the created time field, the function will return the latest incident created time.
+
+* **remove_old_incidents_ids** - Removes old unneccesary incident ids from the last run object to avoid overloading.
+
+* **get_found_incident_ids** - Returns a list of the new fetched incident ids. This is for saving it into the last run object and filter duplicates in the next call of fetch-incidents.
+
+* **create_updated_last_run_object** - Creates a new last run object with a new time and limit for the next fetch.
+
+* **update_last_run_object** - Updates the existing last run object.
```

Comment:
```suggestion
* **filter_incidents_by_duplicates_and_limit()** - After getting the incidents from the 3rd party API call, we need to filter out the duplicate incidents.
In the example above, after an incident A is indexed the fetch will bring incidents A and B, then we must return only A because B was already fetched. In addition, after filtering duplicates, if we still have more incidents than the limit, the function will return only by the limit.

* **get_latest_incident_created_time()** - Given a list of incidents and the created time field, the function will return the latest incident created time.

* **remove_old_incidents_ids()** - Removes old incident IDs from the last run object to avoid overloading.

* **get_found_incident_ids()** - Returns a list of the new fetched incident IDs. This is for saving it into the last run object and filter duplicates in the next call of fetch-incidents.

* **create_updated_last_run_object()** - Creates a new last run object with a new time and limit for the next fetch.

* **update_last_run_object()** - Updates the existing last run object.
```